catalogs/catalog-hadoop/src/main/java/com/datastrato/gravitino/catalog/hadoop/HadoopCatalogOperations.java

```diff
@@ -190,102 +210,122 @@ public Fileset createFileset(
+
+                try {
+                  // formalize the path to avoid path without scheme, uri, authority, etc.
+                  filesetPath = formalizePath(filesetPath, hadoopConf);
+                  FileSystem fs = filesetPath.getFileSystem(hadoopConf);
+                  if (!fs.exists(filesetPath)) {
+                    if (!fs.mkdirs(filesetPath)) {
+                      throw new RuntimeException(
+                          "Failed to create fileset " + ident + " location " + filesetPath);
+                    }
+
+                    LOG.info("Created fileset {} location {}", ident, filesetPath);
+                  } else {
+                    LOG.info("Fileset {} manages the existing location {}", ident, filesetPath);
+                  }
+
+                } catch (IOException ioe) {
+                  throw new RuntimeException(
+                      "Failed to create fileset " + ident + " location " + filesetPath, ioe);
+                }
+
+                StringIdentifier stringId = StringIdentifier.fromProperties(properties);
+                Preconditions.checkArgument(
+                    stringId != null, "Property String identifier should not be null");
+
+                FilesetEntity filesetEntity =
+                    FilesetEntity.builder()
+                        .withName(ident.name())
+                        .withId(stringId.id())
+                        .withNamespace(ident.namespace())
+                        .withComment(comment)
+                        .withFilesetType(type)
+                        // Store the storageLocation to the store. If the "storageLocation" is null
+                        // for
+                        // managed fileset, Gravitino will get and store the location based on the
+                        // catalog/schema's location and store it to the store.
+                        .withStorageLocation(filesetPath.toString())
+                        .withProperties(properties)
+                        .withAuditInfo(
+                            AuditInfo.builder()
+                                .withCreator(apiLoginUser)
+                                .withCreateTime(Instant.now())
+                                .build())
+                        .build();
+
+                try {
+                  store.put(filesetEntity, true /* overwrite */);
+                } catch (IOException ioe) {
+                  throw new RuntimeException("Failed to create fileset " + ident, ioe);
+                }
+
+                return HadoopFileset.builder()
+                    .withName(ident.name())
+                    .withComment(comment)
+                    .withType(type)
+                    .withStorageLocation(filesetPath.toString())
+                    .withProperties(filesetEntity.properties())
+                    .withAuditInfo(filesetEntity.auditInfo())
+                    .build();
+              });
```

Comment:
This method will only throw PrivilegedException, we can refer to my `ProxyPlugin`

Reply:
So what's wrong with the fact that This method will only throw PrivilegedException?